pkg/mod/gomod.go

```diff
@@ -0,0 +1,112 @@
+package mod
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type goModule struct {
+	Path string
+	Dir  string
+}
+
+func goGetByFilepath(filename string) error {
+	dir := filepath.Dir(filename)
+	re := regexp.MustCompile(`^\.|\.\.|/$`)
+
+	for !re.MatchString(dir) {
+		logrus.Debug(dir)
+		err := goGet(dir)
```

Comment:
where does `go get` dump the new files?

Reply:
What `go get` does is downloading new files to `$GOPATH/pkg/mod` and rewriting `go.mod` & `go.sum`.

Reply:
We should change the GOPATH environment variable of the `go get` call so we control the location. i.e the `.sysl/ ` directory if found, or the `SYSLPATH` envvar